cpp/test/core/apply_tests.cpp

```diff
@@ -134,6 +141,210 @@ Bonded(uint16_t version = bond::v1)
 }
 
 
+class IDLBuilder : public bond::Transform
+{
+public:
+    explicit IDLBuilder(std::map<std::string, std::string>& structs)
+        : _structs{ &structs },
+          _qualifiedName{},
+          _container{ bond::BT_UNAVAILABLE }
+    {}
+
+    void Begin(const bond::Metadata& metadata) const
+    {
+        _qualifiedName = &metadata.qualified_name;
+
+        if (_structs->find(*_qualifiedName) == _structs->end())
+            _this << std::boolalpha << "struct " << *_qualifiedName << "\n{\n";
+        else
+            _structs = {};
+    }
+
+    template <typename T>
+    bool Base(const T& value) const
+    {
+        if (!_structs)
+            return true;
+
+        _this.seekp(-3, std::ios_base::end);
```

Comment:
Minor: a comment for the -3 would be nice. 'Removing "\\n{\\n" to add base'?

Reply:
Yep, will update.